src/main/kotlin/com/exactpro/th2/check1/rule/sequence/SilenceCheckTask.kt

```diff
@@ -0,0 +1,152 @@
+import com.exactpro.th2.check1.rule.AbstractCheckTask
+import com.exactpro.th2.check1.rule.ComparisonContainer
+import com.exactpro.th2.check1.rule.MessageContainer
+import com.exactpro.th2.check1.rule.SailfishFilter
+import com.exactpro.th2.check1.util.VerificationUtil
+import com.exactpro.th2.check1.utils.toRootMessageFilter
+import com.exactpro.th2.common.event.Event
+import com.exactpro.th2.common.event.EventUtils.createMessageBean
+import com.exactpro.th2.common.grpc.EventBatch
+import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.common.grpc.RootMessageFilter
+import com.exactpro.th2.common.message.toJson
+import com.exactpro.th2.common.message.toReadableBodyCollection
+import com.exactpro.th2.common.schema.message.MessageRouter
+import io.reactivex.Observable
+import java.time.Instant
+
+class SilenceCheckTask(
+    protoPreFilter: PreFilter,
+    description: String?,
+    taskTimeout: TaskTimeout,
+    maxEventBatchContentSize: Int,
+    submitTime: Instant,
+    sessionKey: SessionKey,
+    parentEventID: EventID,
+    messageStream: Observable<StreamContainer>,
+    eventBatchRouter: MessageRouter<EventBatch>
+) : AbstractCheckTask(
+    description,
+    taskTimeout,
+    maxEventBatchContentSize,
+    submitTime,
+    sessionKey,
+    parentEventID,
+    messageStream,
+    eventBatchRouter
+) {
+    private val protoPreMessageFilter: RootMessageFilter = protoPreFilter.toRootMessageFilter()
+    private val messagePreFilter = SailfishFilter(
+        converter.fromProtoPreFilter(protoPreMessageFilter),
+        protoPreMessageFilter.toCompareSettings()
+    )
+    private val metadataPreFilter: SailfishFilter? = protoPreMessageFilter.metadataFilterOrNull()?.let {
+        SailfishFilter(
+            converter.fromMetadataFilter(it, VerificationUtil.METADATA_MESSAGE_NAME),
+            it.toComparisonSettings()
+        )
+    }
+    private lateinit var preFilterEvent: Event
+    private lateinit var resultEvent: Event
+    private var extraMessagesCounter: Int = 0
+
+    @Volatile
+    private var started = false
+
+    override fun onStart() {
+        super.onStart()
+        started = true
+        val hasNextTask = hasNextTask()
+        if (isParentCompleted == false || hasNextTask) {
+            if (hasNextTask) {
+                LOGGER.info("Has subscribed task. Skip checking extra messages")
+            } else {
+                LOGGER.info("Parent task was not finished normally. Skip checking extra messages")
+            }
+            cancel()
+            checkComplete()
+            return
+        }
+        preFilterEvent = Event.start()
+            .type("preFiltering")
+            .bodyData(protoPreMessageFilter.toReadableBodyCollection())
+
+        rootEvent.addSubEvent(preFilterEvent)
+
+        resultEvent = Event.start()
+            .type("noMessagesCheckResult")
+        rootEvent.addSubEvent(resultEvent)
+    }
+
+    override fun onChainedTaskSubscription() {
+        if (started) {
```

Comment:
This check doesn't guarantee that the start method is called at this moment and will be continued after this check.
I think we should call cancel and checkComplete methods always and start method shouldn't be changed